src/pages/PlayAndLearn/StartLearn.js

```diff
@@ -82,127 +79,188 @@ function StartLearn() {
+            if (tempContent.length > 0) {
+              let getitem = localStorage.getItem('content_random_id')
+                ? localStorage.getItem('content_random_id')
+                : 0;
+              if (trysame != 'yes') {
+                let old_getitem = getitem;
+                while (old_getitem == getitem) {
+                  getitem = randomIntFromInterval(
+                    0,
+                    Number(tempContent.length - 1)
+                  );
+                }
+              }
+              localStorage.setItem('trysame', 'no');
+              localStorage.setItem('content_random_id', getitem);
+              set_content(tempContent[getitem].content);
+              set_content_id(getitem);
+              localStorage.setItem(
+                'contentText',
+                tempContent[getitem].content[
+                  localStorage.getItem('apphomelang')
+                ].text
+              );
             }
-            localStorage.setItem('trysame', 'no');
-            localStorage.setItem('content_random_id', getitem);
-            set_content(tempContent[getitem].content);
-            set_content_id(getitem);
-            localStorage.setItem('contentText', tempContent[getitem].content[localStorage.getItem('apphomelang')].text);
+            scroll_to_top('smooth');
+            setNewTempContent(tempContent);
+            set_load_cnt(load_cnt => Number(load_cnt + 1));
           }
-          scroll_to_top('smooth');
-            setNewTempContent(tempContent)
-          set_load_cnt(load_cnt => Number(load_cnt + 1));
-        }
-      })
-      .catch(err => console.log(err));
+        })
+        .catch(err => console.log(err));
     } else {
       axios
-      .get(window.location.origin + '/db/playAndLearn/proto4.json')
-      .then(res => {
-        localStorage.setItem('contents', JSON.stringify(res.data));
-
-        if (load_cnt === 0) {
-          const content_list = getContentList();
-
-          let tempContent = [];
-          const content_keys = Object.keys(content_list);
-
-          content_keys.forEach(key => {
-            if (
-              content_list[key].type === sel_level &&
-              content_list[key]?.[sel_lang]
-            ) {
-              tempContent.push({
-                content: content_list[key],
-              });
-            }
-          });
-          if (tempContent.length > 0) {
-            let getitem = localStorage.getItem('content_random_id')
-              ? localStorage.getItem('content_random_id')
-              : 0;
-            if (trysame != 'yes') {
-              let old_getitem = getitem;
-              while (old_getitem == getitem) {
-                getitem = randomIntFromInterval(
-                  0,
-                  Number(tempContent.length - 1)
-                );
+        .get(window.location.origin + '/db/playAndLearn/proto4.json')
+        .then(res => {
+          localStorage.setItem('proto4', JSON.stringify(res.data));
```

Comment:
when we are clearing this localstorage for this variables?

Reply:
it can be hande in angular when user is loggin out 
we need to clear all localstorage items